core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -453,7 +462,7 @@ private[log] class Cleaner(val id: Int,
           if (writeOriginalMessageSet)
             ByteBufferMessageSet.writeMessage(writeBuffer, entry.message, entry.offset)
           else if (retainedMessages.nonEmpty)
-            compressMessages(writeBuffer, entry.message.compressionCodec, messageFormatVersion, retainedMessages)
+            compressMessages(writeBuffer, targetCodec, messageFormatVersion, retainedMessages)
```

Comment:
If I go from a more compact compression type (gzip) to a less compact type like (snappy) does that risk resulting in messages that are larger than max.message.bytes?